zerver/lib/streams.py

```diff
@@ -281,7 +281,7 @@ def list_to_streams(streams_raw: Iterable[Mapping[str, Any]],
         created_streams = []  # type: List[Stream]
     else:
         # autocreate=True path starts here
-        if not user_profile.can_create_streams():
+        if not user_profile.has_permission('create_stream_policy'):
```

Comment:
I think for readability, it might be better to keep `.can_create_streams()` and friends as wrappers around `.has_permission`; the old code is a lot more readable in most places where we don't have a dynamic setting.

Reply:
@timabbott do you mean something like this-
```
def has_permission(self, policy_name: str) -> bool:
        
        if policy_name == 'create_stream_policy':
            return self.can_create_streams()

        if policy_name == 'invite_to_stream_policy':
            return self.can_subscribe_other_users()

        raise AssertionError("Invalid policy")
```
and `can_create_streams` ans similar methods remain uncahnged.